token/program-2022/src/extension/mod.rs

```diff
@@ -373,6 +373,38 @@ impl<'data, S: BaseState> StateWithExtensionsMut<'data, S> {
         self.init_or_get_extension(true)
     }
 
+    /// Packs the relevant Extension of an ExtensionType into an open slot if not already
+    /// found in the data buffer, otherwise overwrites the Extension
+    pub fn init_extension_from_type(
+        &mut self,
+        extension_type: ExtensionType,
+    ) -> Result<(), ProgramError> {
+        match extension_type {
```

Comment:
Since this is focused for account initialization, I'm thinking it should only do something for the Account ones, ie. just `TransferFeeConfig` and `AccountPaddingTest`, what do you think?

Reply:
Just depends whether this would be useful for Mint initialization too... I'm guessing not, since Mint extension inits will have their own instructions?

Reply:
Yeah that's the idea right now. I'd prefer to limit this dynamic stuff to avoid any potential vulnerabilities.

Reply:
sg